packages/cw3/examples/schema.rs

```diff
@@ -0,0 +1,23 @@
+use std::env::current_dir;
+use std::fs::create_dir_all;
+
+use cosmwasm_schema::{export_schema, export_schema_with_title, remove_schemas, schema_for};
+
+use cw3::{
+    Cw3HandleMsg, Cw3QueryMsg, ProposalListResponse, ProposalResponse, VoteListResponse,
+    VoteResponse,
+};
+
+fn main() {
+    let mut out_dir = current_dir().unwrap();
+    out_dir.push("schema");
+    create_dir_all(&out_dir).unwrap();
+    remove_schemas(&out_dir).unwrap();
+
+    export_schema_with_title(&mut schema_for!(Cw3HandleMsg), &out_dir, "HandleMsg");
+    export_schema_with_title(&mut schema_for!(Cw3QueryMsg), &out_dir, "QueryMsg");
+    export_schema(&schema_for!(ProposalResponse), &out_dir);
+    export_schema(&schema_for!(ProposalListResponse), &out_dir);
+    export_schema(&schema_for!(VoteResponse), &out_dir);
+    export_schema(&schema_for!(VoteListResponse), &out_dir);
```

Comment:
All those schemas are only generated for `T = Empty`, so `T` cannot be used together with the schema files. If we want to stick with the custom `T` in `CosmosMsg`, I think we should remove the schemas here. Then only the implementations generate schemas, which either set `T` or actually use `Empty`.

(the same is true for the schemas in cosmwasm-std (packages/std/schema), which are kind of pointless)

Reply:
These work for generic queriers that don't support custom messages, and are actually the basis of most generic clients.

I would expect some custom contracts to add more and then we have custom clients that know that, but those would be chain-specific naturally, and this is the only set of schemas that would work over all contracts and all chains.